### ISSUE TITLE
Synchronously call function

### ISSUE THREAD

**Issue by productfrontenddeveloper** [enesozturk/react-native-hold-menu#10]
<img width="628" alt="Screen Shot 2021-03-09 at 20 07 30" src="https://user-images.githubusercontent.com/38668796/110509420-23357600-8113-11eb-9149-b5f863e0e0c9.png">

![Screen Shot 2021-03-09 at 20 08 58](https://user-images.githubusercontent.com/38668796/110509566-49f3ac80-8113-11eb-83b9-54de66e56135.png)

_Originally posted by @productfrontenddeveloper in https://github.com/enesozturk/react-native-hold-menu/issues/9#issuecomment-794175410_

**Comment by enesozturk** [comment]
Could you please share other details in bug issue template for your example? 

`HoldMenuProvide`is should be used in root of your app (App.js) by the way. And what kind a usage are you trying? If you want to use touchable or pressable with HoldMenu you can check [this usage](https://github.com/enesozturk/react-native-hold-menu/issues/2#issuecomment-792221345).

**Comment by productfrontenddeveloper** [comment]
> Are we going to go one by one? Which versions are you using?

oo sorry :( 

```
Package versions

React: 17.0.1
React Native: 0.64.0-rc.4
React Native Reanimated: 2.0.0
```

**Comment by enesozturk** [comment]
And the Hold Menu version? Is it **v0.0.8**? This might because of your Reanimated setup as well. Be sure Reanimated it is working fine.

**Comment by productfrontenddeveloper** [comment]
> And the Hold Menu version? Is it **v0.0.8**? This might because of your Reanimated setup as well. Be sure Reanimated it is working fine.

yes, v0.0.8, Reanimated I've been using it for a long time, but let me check again.

**Comment by enesozturk** [comment]
I see, I have to reproduce problem to solve and will try. Any help would be great to reproduce it.

**Comment by productfrontenddeveloper** [comment]
> I see, I have to reproduce problem to solve and will try. Any help would be great to reproduce it.

Reanimated no problem, add react-native-hold-menu some problem
https://user-images.githubusercontent.com/38668796/110529174-f8a2e780-8129-11eb-90ca-b65ca7e33ea5.mov

**Comment by productfrontenddeveloper** [comment]
https://github.com/software-mansion/react-native-reanimated/releases/tag/2.0.0 

Can it test this with the stable version just released? I wonder if the library does not work steadily

**Comment by enesozturk** [comment]
I have not tested with stable version yet but was working with **rc-3** that last version before the stable. No breaking changes after rc-3 ([maybe this](https://github.com/software-mansion/react-native-reanimated/pull/1639)). Will try it today with 2.0.0.

**Comment by BooYeu** [comment]
@enesozturk Can it run ? I have the same error. I use react-native init to create a project and follow react-native-unimodules installing. Do I need try create-react-native-app to create a project ?

**Comment by enesozturk** [comment]
Hi @BooYeu, I am trying to fix another more important issue for now, after that this must be solved with new release. It is important which React Native and Reanimated version you are using. You can check the [expo project](https://github.com/enesozturk/react-native-hold-menu-expo-example) or clone this repo and run the example project.

**Comment by BooYeu** [comment]
@enesozturk thank you very much ^^ . And I want to know dose it also need expo in the new release ?  My app will be bigger when I choice add react-native-unimodules to enable expo function

**Comment by enesozturk** [comment]
> @enesozturk thank you very much ^^ . And I want to know dose it also need expo in the new release ?  My app will be bigger when I choice add react-native-unimodules to enable expo function

You don't have to use Expo. You can use bare RN setup with unimodules. It is needed for blurview and haptic feedback. We may make it optional further releases.

**Comment by BooYeu** [comment]
@enesozturk ok ok , thank you . I really really like this project and appreciate . And want to know maybe when it can use in react-native@0.64 and react-native-reanimated@2.0.0 stable version ?

**Comment by enesozturk** [comment]
> @enesozturk ok ok , thank you . I really really like this project and appreciate . And want to know maybe when it can use in react-native@0.64 and react-native-reanimated@2.0.0 stable version ?

Not sure. Thanks for let me know the versions. I will check it 👍🏽

**Comment by BooYeu** [comment]
I fix the problem by set _leftOrRight from object to value, and edit "..._leftOrRight" to "left:_leftOrRight"
but now , I found items onPress error . Maybe also because of react-native-reanimated stable version?

**Comment by enesozturk** [comment]
There must be something missing. _leftOrRight is returning an `object` are you sure? And what if it is opening from right? Yes there is an issue I am following it. https://github.com/software-mansion/react-native-reanimated/issues/1758
Reanimated team also worked on int an releases in `v2.0.1` but still exist -> https://github.com/software-mansion/react-native-reanimated/pull/1844

**Comment by BooYeu** [comment]
I see the object _leftOrRight just has one param . So I change _leftOrRight to number, and edit it like "left:_leftOrRight". Then it can run and shows ok. Do you have error when click the item . I give an onPress param , But it shows like this
![41616844434_ pic](https://user-images.githubusercontent.com/16587763/112719173-80faf800-8f32-11eb-812c-a3224ee51a1f.jpg)

**Comment by BooYeu** [comment]
and it seems your doc write a wrong props . holditems your doc shows styles , but it is containerStyles in codes~~

**Comment by enesozturk** [comment]
> and it seems your doc write a wrong props . holditems your doc shows styles , but it is containerStyles in codes~~

Oh I see, will fix it 👍🏽

**Comment by BooYeu** [comment]
and item text prop doesn't support Chinese like " 你好". It shows error code.I review the code the whole day. But I don't know why...Do you have any idea ? Thank you very much !!!!
![11616846767_ pic](https://user-images.githubusercontent.com/16587763/112720200-e56c8600-8f37-11eb-8228-d857e1090d36.jpg)

**Comment by enesozturk** [comment]
That is weird. No idea why. Maybe it is related the same issue. Text is passing through Shared Value. That might breake something.

**Comment by cayasso** [comment]
I am experiencing this exact same error, any fix for it yet?

**Comment by cayasso** [comment]
As @BooYeu pointed out the solution, that works because object spread is not supported in worklets, its a known problem, as mentioned in the reanimated docs here https://docs.swmansion.com/react-native-reanimated/docs/next/#known-problems-and-limitations

**Comment by bfischer1121** [comment]
Maybe you guys could release this to npm? We're having to point to the develop branch because the npm version is broken and I don't think it's going to work for anyone.

**Comment by bfischer1121** [comment]
For anyone encountering this issue, the simplest thing I found is to fork the repo, build the develop branch, remove lib from the .gitignore and commit it, then point to your fork in the package.json

**Comment by riamon-v** [comment]
Having the exaxt same issue here too.
Using react-native: 0.65.1
And last version of this package

**Comment by riamon-v** [comment]
> For anyone encountering this issue, the simplest thing I found is to fork the repo, build the develop branch, remove lib from the .gitignore and commit it, then point to your fork in the package.json

Not working for me
